digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/IntegrationApi.scala

```diff
@@ -240,6 +246,33 @@ class IntegrationApi(val massTransitStopService: MassTransitStopService) extends
         .getOrElse("")
   }
 
+  def lastModifiedBy(createdBy: Option[String], modifiedBy: Option[String]): (String, Boolean) = {
+
```

Comment:
Instead of having the regex validation you cold have something like that

```
def nonFixedUsers(userOption: Option[String])={
      val nonFixedValues = List("split_speedlimit_", "batch_process_")
      userOption match {
        case Some(user) if(nonFixedValues.exists(v => user.startsWith(v))) =>
          user
        case _ =>
          ""
      }
    }
```

And then you can using direclly in the list 

```
val autoGeneratedValues = List("dr1conversion", "dr1_conversion", nonFixedUsers(modifiedBy), nonFixedUsers(createdBy), "automatic_correction", "excel_data_migration", "automatic_generation", "vvh_generated")
```